zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -239,16 +250,7 @@ protected ZkClient(IZkConnection zkConnection, int connectionTimeout, long opera
     } else {
       LOG.info("ZkClient monitor key or type is not provided. Skip monitoring.");
     }
-
-    connect(connectionTimeout, this);
-
-    try {
-      if (_monitor != null) {
-        _monitor.register();
-      }
-    } catch (JMException e){
-      LOG.error("Error in creating ZkClientMonitor", e);
-    }
+    _watcher = watcher == null ? this : watcher;
```

Comment:
neat : We should add comment here. If I understand correctly then either "Raw ZkClient" passes "watcher" in "native Zkclient"'s instantiation or legacy code instantiates native zkclient without any watcher. If that's the case then we should add comments at bot places (native zk client and raw zk client) IMHO.

Reply:
updated, thanks!